fix: Support Notion v8 UUIDs

ToriK17 opened the pull request:
## Fix: Support Notion v8 UUIDs
This PR fixes an issue where valid Notion block IDs (e.g. version 8 UUIDs) would be rejected by the uuid.validate() check, resulting in misleading errors like:

```Invalid URL or blockId: https://www.notion.so/Whatever-1cf62d960d7f80c79960c58edb3217fd```
### The Cause
Notion block IDs look like regular UUIDs — but they can include version 8, which is outside the RFC4122 spec (1-5). For example:
Your Notion block ID might look like this:
```1cf62d96-0d7f-80c7-9960-c58edb3217fd```
That third segment is 80c7.
The uuid.validate() method from older versions didn’t recognize this as valid.

### Solution
We simply update the uuid dependency to ^11.1.0, which does validate v8 UUIDs correctly. 

Here’s the notion URL that led to this finding:
```URL: https://www.notion.so/Network-Engineering-1cf62d960d7f80c79960c58edb3217fd```

### Tests
- Added three test cases for a Notion-style v8 UUID

ToriK17 committed:
allow UUIDS from notion that are any characters

yannbolliger commented:
Hey there, thanks a lot for your interest in the notion-exporter ✨ 

The code changes are appreciated very much. However, while checking the README of the `uuid` package I saw [the following](https://github.com/uuidjs/uuid?tab=readme-ov-file#uuidv8):

> The RFC does not define a creation algorithm for them [uuid v8], which is why this package does not offer a v8() method. The validate() and version() methods do work with such UUIDs, however.

Therefore the question: would it also work to update the `uuid` dependency and then use the old implementation? (Also, let's add a test case for the example you provided above, please 🙏)

ToriK17 committed:
update UUID package to handle v8 ids

ToriK17 commented:
Hey, thanks for the quick reply! 
You're absolutely right — uuid@11.x does support v8 UUIDs in the validate() method. That was a great catch, and updating the dependency did resolve the issue without needing a custom validator.
I've reverted the custom function and just bumped the uuid version instead. Also added test coverage for the v8 format I encountered from Notion.
Totally open to further edits if needed — and truly, thanks for building and sharing this package. It saved me hours (after those UUIDs cost me some first 😅).

yannbolliger commented on test/blockId.test.ts:
```diff
@@ -19,6 +19,13 @@ describe("Validate UUID with and without dashes", () => {
     ).toBeUndefined()
   })
 
+  it("Validates v8 UUIDs (used by Notion)", () => {
+    // This UUID has a version 8 characters (8xxx in the third group)
+    expect(validateUuid("1cf62d960d7f80c79960c58edb3217fd")).toEqual(
+      "1cf62d96-0d7f-80c7-9960-c58edb3217fd"
+    )
+  })
```
The only thing, we could _optionally_ add would be a second test with a v8 UUID that already has dashes...

ToriK17 replied:
That is a solid suggestion! I added in a test for the dashed v8 UUID too. Appreciate the review and had fun contributing to this project 😄

yannbolliger reviewed (approved):
Yeahy, this is great. Thanks for your contribution 👍

yannbolliger commented:
Oh and you'll have to run [prettier](https://prettier.io/) over the changed files to make the CI formatting check happy.

ToriK17 committed:
add aditional test for v8 UID including dashes format

ToriK17 committed:
Merge branch 'master' into fix-blockid-validation
